macros/fivetran/base/fivetran_fb_url_tag.sql

```diff
@@ -0,0 +1,81 @@
+{% macro fivetran_fb_url_tag() %}
+
+    {{adapter_macro('facebook_ads.fivetran_fb_url_tag')}}
+
+{% endmacro %}
+
+{% macro default__fivetran_fb_url_tag() %}
+
+with base as (
+    
+select distinct
+    type,
+    key,
+    creative_id,
+    value
+
+from
+    {{ var('url_tag_table') }}
+  
+),
+
+pivot as (
+    
+    select 
+
+        creative_id,
+    
+        {% for key, value in {
```

Comment:
use a set at the top of the model to create the map and then iterate over it here in the body of the model.